corehq/apps/users/models.py

```diff
@@ -3003,6 +3003,36 @@ class UserOptionToggles(models.Model):
     use_latest_build_cloudcare = models.BooleanField(default=False)
 
 
+def is_option_enabled(option, username=None, domain_name=None):
```

Comment:
I didn't anticipate this function when reviewing the new models - this seems like the core of this new paradigm, is that right?  That is, two intrinsically linked models that have the same settings so things can be configured at the user or domain level.  Is there anything that could be done to make the linkage between the two more rigid, like enforcing that they have the same set of fields?

Reply:
(Partly answered in the comments below, and partly avoided in the new design.) 

If I were to use this approach again, I would put the models in the same module, and add unit tests and maybe a doctest to show possible scenarios.

They are meant to mimic toggles. So a domain-only toggle would only have an option defined in `DomainOptionToggles`, a user-only toggle would only have an option defined in `UserOptionToggles`, and a toggle that can be enabled for both would be defined in both models (like the one in this change). Like toggles, all options are booleans that default to false, and are true if they are true for the domain or the user.